[pytorch] fix backend_type for backend/PG plugin

kingchc opened the pull request:
Summary: For backend/PG plugin, use `ProcessGroup.BackendType.CUSTOM` to avoid uninitialized variable during `pg._register_backend` later

Test Plan: CI/CD and internal tests

Differential Revision: D42793222

H-Huang reviewed (approved):
LGTM, thanks for adding. Please sign the CLA to fix that failing check!

Commit:
[pytorch] fix backend_type for backend/PG plugin (#93129)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/93129

For backend/PG plugin, use `ProcessGroup.BackendType.CUSTOM` to avoid uninitialized variable during `pg._register_backend` later

Test Plan: CI/CD and internal tests

Reviewed By: H-Huang

Differential Revision: D42793222

fbshipit-source-id: c7f4cfd5d829c6ee606b26c3496f955019a0c854
